backend/clap/src/main/java/zpi/ppea/clap/service/SubmissionService.java

```diff
@@ -0,0 +1,43 @@
+package zpi.ppea.clap.service;
+
+import data_store.*;
+import lombok.RequiredArgsConstructor;
+import net.devh.boot.grpc.client.inject.GrpcClient;
+import org.springframework.stereotype.Service;
+import zpi.ppea.clap.config.ValueConfig;
+import zpi.ppea.clap.dtos.DetailsSubmissionResponseDto;
+import zpi.ppea.clap.dtos.SubmissionDto;
+import zpi.ppea.clap.logic.BusinessLogicService;
+import zpi.ppea.clap.mappers.DetailedSubmissionMapper;
+import zpi.ppea.clap.mappers.SubmissionMapper;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class SubmissionService {
+    @GrpcClient("dataStore")
+    DataStoreGrpc.DataStoreBlockingStub dataStoreBlockingStub;
```

Comment:
would be nice if we had a cache. I recommend this one `https://github.com/ben-manes/caffeine` i used it before and its easy to use

Reply:
While cache is nice to have, I think we should think about it later, as we're currently quite tight on time. And handling cache invalidation might cause more issues to us.

Reply:
Your choice, but as Bartek said, we're in a hurry. I can take it but we still have a lot to do.